src/main/java/taxi/dao/jdbc/CarJdbcDao.java

```diff
@@ -0,0 +1,165 @@
+package taxi.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.CarDao;
+import taxi.lib.Dao;
+import taxi.model.Car;
+import taxi.model.Driver;
+import taxi.model.Manufacturer;
+import taxi.services.util.ConnectionUtil;
+
+@Dao
+public class CarJdbcDao implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create car" + car.getModel(), e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getCarQuery = "SELECT * FROM cars WHERE id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getCarQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.ofNullable(getCarFromResult(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant get car by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String getAllCars = "SELECT * FROM cars WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getAllCars)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResult(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create list of cars", e);
+        }
```

Comment:
ты не сетишь drivers в машину, сделай отдельный метод, например getDrivers(long carId) с query, который будет выдавать тебе всех драйверов по каждой машине
```suggestion
        List<Car> cars = new ArrayList<>();
        String getAllCars = "SELECT * FROM cars WHERE deleted = false";
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement preparedStatement = connection.prepareStatement(getAllCars)) {
            ResultSet resultSet = preparedStatement.executeQuery();
            while (resultSet.next()) {
                cars.add(getCarFromResult(resultSet));
            }
        } catch (SQLException e) {
            throw new RuntimeException("Can't create list of cars", e);
        }
        for (Car car : cars) {
        car.setDrivers(getDrivers(car.getId()));
        }
        return cars;
```

Reply:
так еще в методе getAllByDriverId() и get()